### PR TITLE
Upgrade lambda node version from 8 to 12

### LINKED ISSUE #32: SLS: update runtime to node12
Let's update our runtime!

- [x] Update serverless runtime to node 12 (or 10 if issues).

### PR DESCRIPTION
**Pull request by stevenmusumeche**
Fixes #32

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,10 @@
   "repository": "https://github.com/FormidableLabs/aws-lambda-serverless-reference",
   "author": "Ryan Roemer <ryan.roemer@formidable.com>",
   "license": "MIT",
+  "engines": {    
+    "node": "~12.14.0"
```
Comment (ryan-roemer) [inline package.json]: This reference project _could_ be installed on lower nodes that are still supported by AWS. Let's remove the restriction and `engine-strict` maybe?

**Review by ryan-roemer — COMMENTED** [review]
Nice work! Have one suggested change.

When this is ready for merging, can you deploy fully to sandbox and post the base URL here for us to kick the tires on? Thanks!

File: package.json (hunk 1 above)
Comment (stevenmusumeche) [reply package.json]: node 8 is EOL on lambda next week.  Can we restrict to > 10?

**Comment by stevenmusumeche** [comment]
```Service Information
service: sls-simple-reference
stage: sandbox
region: us-east-1
stack: sls-simple-reference-sandbox
resources: 50
api keys:
  None
endpoints:
  ANY - https://amx4dqzs4j.execute-api.us-east-1.amazonaws.com/sandbox/base
  ANY - https://amx4dqzs4j.execute-api.us-east-1.amazonaws.com/sandbox/base/{proxy+}
  ANY - https://amx4dqzs4j.execute-api.us-east-1.amazonaws.com/sandbox/xray
  ANY - https://amx4dqzs4j.execute-api.us-east-1.amazonaws.com/sandbox/xray/{proxy+}
  ANY - https://amx4dqzs4j.execute-api.us-east-1.amazonaws.com/sandbox/vpc
  ANY - https://amx4dqzs4j.execute-api.us-east-1.amazonaws.com/sandbox/vpc/{proxy+}
  ANY - https://amx4dqzs4j.execute-api.us-east-1.amazonaws.com/sandbox/canary
  ANY - https://amx4dqzs4j.execute-api.us-east-1.amazonaws.com/sandbox/canary/{proxy+}
  ANY - https://amx4dqzs4j.execute-api.us-east-1.amazonaws.com/sandbox/layers
  ANY - https://amx4dqzs4j.execute-api.us-east-1.amazonaws.com/sandbox/layers/{proxy+}
functions:
  base: sls-simple-reference-sandbox-base
  xray: sls-simple-reference-sandbox-xray
  vpc: sls-simple-reference-sandbox-vpc
  canary: sls-simple-reference-sandbox-canary
  layers: sls-simple-reference-sandbox-layers
layers:
  vendor: arn:aws:lambda:us-east-1:819013376994:layer:sls-simple-reference-sandbox-vendor:41
  repeat: arn:aws:lambda:us-east-1:819013376994:layer:sls-simple-reference-sandbox-repeat:44

File: package.json (hunk 1 above)
Comment (ryan-roemer) [reply package.json]: This is an upstream app. I'd rather just leave it to `.nvmrc` than placing an enforcement here. If this was a lib and we knew we only had node10 code, then sure. But I don't see good value in an arbitrarily high enforcement...

File: package.json [hunk 2]
```diff
@@ -5,6 +5,9 @@
   "repository": "https://github.com/FormidableLabs/aws-lambda-serverless-reference",
   "author": "Ryan Roemer <ryan.roemer@formidable.com>",
   "license": "MIT",
+  "engines": {    
+    "node": ">=10"
```
Comment (ryan-roemer) [inline package.json]: (picking up on actual visible code thread) My vote here is remove the `engines` until we use something that is _actually_ node10+ and use `.nvmrc` as our best practice means of getting folks on the right engine. AWS will still support create + update node8 for a few more months and nothing has changed in our very node8 friendly code...

... but I'm not going to block on this if you have consensus from the backend devs team and feel it's a move we should take here.

**Review by ryan-roemer — APPROVED** [review]
Nice work! Thanks for the output report!

One final comment that we can take or leave...

File: package.json (hunk 1 above)
Comment (ryan-roemer) [reply package.json]: I'm resolving this convo and picking it up on the actual living code.

File: package.json (hunk 2 above)
Comment (stevenmusumeche) [reply package.json]: sounds good, i'll remove it

**Pull request merged**